### ISSUE TITLE
DPS device client fails to provision group registration with invalid URI characters (e.g. colon) over HTTP

### ISSUE THREAD

**Issue by drwill-ms** [Azure/azure-iot-sdk-csharp#3061]
When using the following sample [azure-iot-sdk-csharp/provisioning/device/samples/how to guides/SymmetricKeySample at main · Azure/azure-iot-sdk-csharp · GitHub](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/provisioning/device/samples/how%20to%20guides/SymmetricKeySample)
The Primary key calculated via: [azure-iot-sdk-csharp/provisioning/device/samples/getting started/ComputeDerivedSymmetricKeySample at main · Azure/azure-iot-sdk-csharp · GitHub](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/provisioning/device/samples/getting%20started/ComputeDerivedSymmetricKeySample)

When trying to provision a device using a registration ID that contains the ":" character, it does not work and receive the following error:

```
"Microsoft.Azure.Devices.Provisioning.Client.ProvisioningTransportException: {"errorCode":401002,"trackingId":"xxx","message":"Unauthorized","timestampUtc":"2022-12-12T11:48:20.2787818Z"}
---> Microsoft.Rest.HttpOperationException: Operation returned an invalid status code 'Unauthorized'
at Microsoft.Azure.Devices.Provisioning.Client.Transport.RuntimeRegistration.RegisterDeviceWithHttpMessagesAsync(String registrationId, String idScope, DeviceRegistration deviceRegistration, Nullable`1 forceRegistration, Dictionary`2 customHeaders, CancellationToken cancellationToken)
at Microsoft.Azure.Devices.Provisioning.Client.Transport.RuntimeRegistrationExtensions.RegisterDeviceAsync(IRuntimeRegistration operations, String registrationId, String idScope, DeviceRegistration deviceRegistration, Nullable`1 forceRegistration, CancellationToken cancellationToken)
at Microsoft.Azure.Devices.Provisioning.Client.Transport.ProvisioningTransportHandlerHttp.RegisterAsync(ProvisioningTransportRegisterMessage message, CancellationToken cancellationToken)
--- End of inner exception stack trace ---
at Microsoft.Azure.Devices.Provisioning.Client.Transport.ProvisioningTransportHandlerHttp.RegisterAsync(ProvisioningTransportRegisterMessage message, CancellationToken cancellationToken)
at Microsoft.Azure.Devices.Provisioning.Client.Samples.ProvisioningDeviceClientSample.RunSampleAsync() in /azure-iot-sdk-csharp/provisioning/device/samples/how to guides/SymmetricKeySample/ProvisioningDeviceClientSample.cs:line 49"
```

**Comment by drwill-ms** [comment]
Fixed in main and latest LTS branch. LTS release is pending and will be included in next (yet unscheduled) release from main.

**Comment by drwill-ms** [comment]
Fixed bug in Microsoft.Azure.Devices.Provisioning.Http NuGet with patched LTS version [1.12.5](https://www.nuget.org/packages/Microsoft.Azure.Devices.Provisioning.Transport.Http/1.12.5) of the provisioning transport/HTTP NuGet.

Other relevant, corresponding LTS patch release NuGet versions include:

- Microsoft.Azure.Devices.Client 1.36.7
- Microsoft.Azure.Devices.Shared 1.27.2
- Microsoft.Azure.Devices.Provisoning.Client 1.16.5

See [patch notes](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/lts_2021-3-18_patch7) for more details.
